src/services/IGameContext.hh

```diff
@@ -0,0 +1,42 @@
+#ifndef RA_SERVICES_IGAMECONTEXT_HH
+#define RA_SERVICES_IGAMECONTEXT_HH
+#pragma once
+
+#include <string>
+
+namespace ra {
+namespace services {
+
+class IGameContext
+{
+public:
+    /// <summary>
+    /// Loads the data for the specified game id.
+    /// </summary>
+    virtual void LoadGame(unsigned int nGameId, const std::wstring& sGameTitle) = 0;
+
+    /// <summary>
+    /// Gets the unique identifier of the currently loaded game.
+    /// </summary>
+    virtual unsigned int GameId() const = 0;
+
+    /// <summary>
+    /// Gets the title of the currently loaded game.
+    /// </summary>
+    virtual const std::wstring& GameTitle() const = 0;
+
+    /// <summary>
+    /// Gets the hash of the currently loaded game.
+    /// </summary>
+    virtual const std::string& GameHash() const = 0;
+    
+    /// <summary>
+    /// Sets the game hash.
+    /// </summary>
+    virtual void SetGameHash(const std::string& sGameHash) = 0;
```

Comment:
I don't think getters and setters should be part of an interface, if you absolutely need them, they should be in the implementation. One of the main reasons is so you can make them a constant expression if the field used is a literal type (able to be used in constant expressions like primitives).

Reply:
The getters are the interface. This object stores data about the currently loaded game. `LoadGame` and `SetGameHash` are the initialization methods. At some point in the future, `LoadGame` will make the API call to fetch the patch data, which will populate the other getters. Similarly, resolving the game hash to an ID might get moved into this service as well.

Another thing that I plan to do in a future PR is add a callback so viewmodel can subscribe to when the game data changes, which is currently handled by multiple locations explicitly informing each window that cares.

As such, this class behaves more like a view model than a service, though it won't actually be bound to any views. I'll eliminate the interface and move the implementation out of impl and register the instance directly (not through the interface). Not having the interface makes it slightly harder to mock, but leaving the backing fields as protected should allow for direct setting via a subclass instead of having to mock the entire Load process.